install/installer/pkg/components/spicedb/secret.go

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License.AGPL.txt in the project root for license information.
+
+package spicedb
+
+import (
+	"fmt"
+
+	"github.com/gitpod-io/gitpod/installer/pkg/common"
+	"github.com/gitpod-io/gitpod/installer/pkg/config/v1/experimental"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func secret(ctx *common.RenderContext) ([]runtime.Object, error) {
+	cfg := getExperimentalSpiceDBConfig(ctx)
+	if cfg == nil {
+		return nil, nil
+	}
+
+	// There's a secret-ref provided, we do not need to create a secret
+	if cfg.SecretRef != "" {
+		return nil, nil
+	}
+
+	generated, err := common.RandomString(20)
```

Comment:
Wondering if it's worth keeping this case here, when it might lead to similar confusion as the infamous "loginKey". :right_anger_bubble:

Reply:
We'd only configure this in preview, for other envs I'd expect we make it static